lib/utils/object.ts

```diff
@@ -0,0 +1,215 @@
+import cloneDeepWith from 'lodash/cloneDeepWith';
+import isEqual from 'lodash/eq';
+import forOwn from 'lodash/forOwn';
+import getValue from 'lodash/get';
+import isFunction from 'lodash/isFunction';
+import isPlainObject from 'lodash/isPlainObject';
+import mergeWith from 'lodash/mergeWith';
+import { getComplexKeys } from './format';
+// eslint-disable-next-line import/order -- caused by temporarily mixing require with import
+import { camelize } from './string';
+
+// eslint-disable-next-line @typescript-eslint/no-require-imports -- .js files must be imported using require
+const baseIsNative = require('lodash/_baseIsNative');
+
+/**
+ * Deeply merges object `b` into `a`.
+ * Mutates `a`.
+ *
+ * Same concept as _.merge, but doesn't overwrite properties that have already been assigned.
+ *
+ * @param a
+ * @param b
+ */
+export function mergeDefaults<T>(a: T, b: Partial<T>): T {
+  return mergeWith(a, b, (objectValue, sourceValue) => {
+    // If it's an object, let _ handle it this time, we will be called again for each property
+    if (!isPlainObject(objectValue) && objectValue !== undefined) {
+      // _.isNative includes a check for core-js and throws an error if present.
+      // Depending on _baseIsNative bypasses the core-js check.
```

Comment:
Do you mean _.baseIsNative here (with the period)?

Reply:
It's an internal method from lodash (lodash/_baseIsNative).

Reply:
Ah, gotcha